src/libstd/io/buffered.rs

```diff
@@ -30,7 +31,9 @@ use ptr;
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct BufReader<R> {
     inner: R,
-    buf: Cursor<Vec<u8>>,
+    buf: Vec<u8>,
+    pos: usize,
+    cap: usize,
```

Comment:
I personally tried to keep this to a `Cursor<Vec<u8>>` because it just felt like it was duplicating the functionality otherwise, but I think it would otherwise require unsafe calls to `set_len` which is somewhat worrisome.

I think that the `buf` could be represented as `Box<[u8]>` as well (as the capacity isn't necessary, just the length), but I also think it's fine for now.